src/attestation_types/quote.rs

```diff
@@ -54,6 +71,23 @@ impl Default for CertDataType {
     }
 }
 
+impl TryFrom<u16> for CertDataType {
+    type Error = QuoteError;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        match value {
+            1 => Ok(CertDataType::PpidPlaintext),
+            2 => Ok(CertDataType::PpidRSA2048OAEP),
+            3 => Ok(CertDataType::PpidRSA3072OAEP),
+            4 => Ok(CertDataType::PCKLeafCert),
+            5 => Ok(CertDataType::PCKCertChain),
+            6 => Ok(CertDataType::Quote),
+            7 => Ok(CertDataType::Manifest),
+            _ => Err(QuoteError("Unknown Cert Data type".to_string())),
```

Comment:
You might want to add the `u16` which caused the error to the `QuoteError` as you do in `AttestationKeyType`

Reply:
Good idea.